### ISSUE TITLE
test acc still low

### ISSUE THREAD

**Issue by bikong2** [jeong-tae/DeepFashion-Tensorflow#3]
Hi,
As you mentioned in text, I got a same validation acc as you, but the test acc is really low and sames do not make sense.
Have you got the idea why it work not well?

Another question.
Why feature dimensions extract from vgg16/fc7 is 1536-d?

**Comment by jeong-tae** [comment]
The reason test acc poor is imbalance of dataset.
In validation set, each category are sampled from training dataset which divide equally.
I didn't check the proportion of dataset, but i believe that will be main reason.
You can check and let me know!

The number of feature dimensions are depends on number of labels.
Original vgg16 has 1000 categories, and features should be enough to distinct, at least more than 1000(this is my opinion)
In this code, i didn't use so many number of features. i modified a little bit.

-----------------------------------------------------------------------------------------------------------------------

i was confused your question.
fc7 feature is shared feature that used to predict categories and attributes both.
So when retrieve the images, i think, shared feature will helps.

**Comment by bikong2** [comment]
@jeong-tae 
i checked twice, acc of validate is high because it always predict the same category label, such as 41 or 18 i saw in my test. and for eval, it sames there is not sample from this category, so acc is 0.
it sames the code do not work and i am now doing some debug.

**Comment by jeong-tae** [comment]
Thank you for your feedback!
I will check that issue also. Basically i sampled the data a little(almost 10~15%) randomly. 
So it can be happen.

We can fix data sampling to uniform sampling for each category. then we can get expected results

**Comment by jeong-tae** [comment]
Hi,
 
I am concerning about checking a convergence. But this week is golden week(thanksgiving day) in Korea so that i can't work so soon.
Week day is over at October 9th. If you okay plz wait for it.
 
I am really sorry i couldn't to deal this issue aggressively. 
if you okay send me the graph/plot of loss or accuracy then i can help you by seeing plot during the weekend on cellphone.
I can't use computer on weekend that is reason i can't help you right now.
 
Bests,
JeongTae Lee

**Comment by bikong2** [comment]
@jeong-tae thanks, hope for your replay when you are convenient.

**Comment by jeong-tae** [comment]
Are you waiting for the answer?

I checked the code about this issue. and tried to fix it.
i think to fix this issue will be take a little bit long time. Some categories dominates over few sample categories during the training. briefly, it's biased.

More details are reported soon. Thank you for your cooperation! @bikong2 

I didn't concern about the performance when i made this code. buz i follow the DeepFashion paper that did not publish the code exactly. Just tried making architecture.
If other version of code is published, then i will re-working following that code.
